core/modules/replica/IngestServerConnection.cc

```diff
@@ -481,26 +486,63 @@ void IngestServerConnection::_loadDataIntoTable() {
 
             string const sqlTable = sqlDatabase + "." + h.conn->sqlId(_table);
 
-            statements.push_back(
+            tableMgtStatements.push_back(
                 "ALTER TABLE " + sqlTable + " ADD PARTITION IF NOT EXISTS (PARTITION " + sqlPartition +
                     " VALUES IN (" + to_string(_transactionId) + "))"
             );
-            statements.push_back(
+            dataLoadStatement =
                 "LOAD DATA INFILE " + h.conn->sqlValue(_fileName) +
                     " INTO TABLE " + sqlTable +
-                    " FIELDS TERMINATED BY " + h.conn->sqlValue(string() + _columnSeparator)
-            );
+                    " FIELDS TERMINATED BY " + h.conn->sqlValue(string() + _columnSeparator);
         }
-        for (auto&& statement: statements) {
+        for (auto&& statement: tableMgtStatements) {
             LOGS(_log, LOG_LVL_DEBUG, context << __func__ << "  statement: " << statement);
         }
-        h.conn->execute([&statements](decltype(h.conn) const& conn_) {
-            for (auto&& statement: statements) {
-                conn_->begin();
-                conn_->execute(statement);
-                conn_->commit();
+        LOGS(_log, LOG_LVL_DEBUG, context << __func__ << "  statement: " << dataLoadStatement);
+
+        // Allow retries for the table management statements in case of deadlocks.
+        // Deadlocks may happen when two or many threads are attempting to create
+        // or modify partitioned tables, or at a presence of other threads loading
+        // data into these tables.
+        //
+        // TODO: the experimental limit for the maximum number of retries may need
+        //       to be made unlimited, or be limited by some configurable timeout.
+        int const maxRetries = 1;
```

Comment:
3 may be better?

Reply:
With 1 the problems seems to be gone. I would like to keep it this way to to get some stats on the problem should it happen again. Then I would have to look at more sophisticated lock avoidance techniques.